### PR TITLE
Update [Library],focused_widget

### PR DESCRIPTION
**Pull request by ronso0**
updated documentation for https://github.com/mixxxdj/mixxx/pull/4618

### CODE REVIEW

**Comment by ronso0** [comment]
https://deploy-preview-476--mixxx-manual.netlify.app/chapters/appendix/mixxx_controls.html#control-[Library]-focused_widget

File: source/chapters/appendix/mixxx_controls.rst [hunk 1]
```diff
@@ -3435,21 +3435,27 @@ Note that :mixxx:coref:`[Library],MoveUp` and other Move and Scroll controls emu
...
+     ========  ==========  ================================================================================================
+     Value     writeable   Widget
+     ========  ==========  ================================================================================================
+     0                     none
+     --------  ----------  ------------------------------------------------------------------------------------------------
+     1         x           Search bar
+     --------  ----------  ------------------------------------------------------------------------------------------------
+     2         x           Tree view
+     --------  ----------  ------------------------------------------------------------------------------------------------
+     3         x           Tracks table or root views of library features
```
Comment (Swiftb0y) [inline source/chapters/appendix/mixxx_controls.rst:3450]: We used uppercase X in all other tables.
```suggestion
     1         X           Search bar
     --------  ----------  ------------------------------------------------------------------------------------------------
     2         X           Tree view
     --------  ----------  ------------------------------------------------------------------------------------------------
     3         X           Tracks table or root views of library features
```

**Review by Swiftb0y — CHANGES_REQUESTED** [review]
lgtm otherwise

**Comment by ronso0** [comment]
Comment addessed, and `GoToItem` updated as well.

**Review by Swiftb0y — APPROVED** [review]

**Comment by ronso0** [comment]
merge?

**Review by Swiftb0y — COMMENTED** [review]
Sorry I had a thought shortly after approving but forgot to actually submit the review.

File: source/chapters/appendix/mixxx_controls.rst [hunk 2]
```diff
@@ -3435,21 +3435,27 @@ Note that :mixxx:coref:`[Library],MoveUp` and other Move and Scroll controls emu
...
+     2         X           Tree view
+     --------  ----------  ------------------------------------------------------------------------------------------------
+     3         X           Tracks table or root views of library features
+     --------  ----------  ------------------------------------------------------------------------------------------------
+     4                     Context menu (menus of library widgets or other editable widgets, or main menu bar)
+     --------  ----------  ------------------------------------------------------------------------------------------------
+     5                     Dialog (any confirmation or error popup, preferences, track properties or cover art window)
+     --------  ----------  ------------------------------------------------------------------------------------------------
+     6                     Unknown (widgets that don't fit into any of the above categories)
+     ========  ==========  ================================================================================================
```
[the remark this answers is not in the document]
Comment (ronso0) [reply source/chapters/appendix/mixxx_controls.rst:3457]: The idea is 0 is null/none (self-explaining), and then there is the 1-3 block with writeable widgets from the previous approach. Breaking up that most meaningful sequence of 0-4 doesn't make much sense to me tbh.

[the remark this answers is not in the document]
Comment (Swiftb0y) [reply source/chapters/appendix/mixxx_controls.rst:3457]: Well, assuming we add more widgets, they will have to be added after the `Unknown` category/slot. Which also does not make much sense semantically. If you think that having 0-4 grouped together makes sense, then I'd propose to shift them up and make `Unknown` the zero slot.

[the remark this answers is not in the document]
Comment (Swiftb0y) [reply source/chapters/appendix/mixxx_controls.rst:3457]: Having `Unknown` as 0 also makes sense from the POV that its like the `nullptr` when we simply need an implicit fallback value...

To be honest, I'm not super fixated on this issue I just thought I'd mention it. Its fine if you reject it when you don't see the benefit. Saves a little bit of work too.

[the remark this answers is not in the document]
Comment (ronso0) [reply source/chapters/appendix/mixxx_controls.rst:3457]: Okay, thanks for your review then.
I just don't see use cases for more concrete _FocusWidget_ enum items:
* there are no (focusable) widgets that need to be considered in scripts, all have dedicated controls already (e.g. spinboxes and effect selectors)
* I doubt we add more focusable widgets to the current skin system
* for a future multi-pane library the enum concept is not suitable and we'd to rework the focus concept anyway

[the remark this answers is not in the document]
Comment (ronso0) [reply source/chapters/appendix/mixxx_controls.rst:3457]: :bell:  _ping_ @Swiftb0y ?

[the remark this answers is not in the document]
Comment (Swiftb0y) [reply source/chapters/appendix/mixxx_controls.rst:3457]: Sorry for neglecting this. Thanks for the ping. I don't have any more arguments to convince you. Though I don't consider "we won't change this anyways" a very foresightful perspective.
I won't die on this hill, and the current state is already implemented, so lets just keep it as is.

**Review by Swiftb0y — APPROVED** [review]

File: source/chapters/appendix/mixxx_controls.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (ronso0) [reply source/chapters/appendix/mixxx_controls.rst:3457]: > * I doubt we add more focusable widgets to the current skin system
> * for a future multi-pane library the enum concept is not suitable and we'd need to rework the focus concept anyway

Sorry if this sounds arrogant or ignorant, it's just that I don't see that anyone would invest more time into significant additions the current skin engine or work on a multi-pane library.

Also, please take a look at the `getFocusWidget` implementation https://github.com/mixxxdj/mixxx/blob/main/src/library/librarycontrol.cpp#L766. We already fallback to `None` if no Mixxx window has focus or if the main window has no focused widget, and `None` is considered in all library navigation slots.

`[Libary], focused_widget` is in main only and that is still alpha, so if you think we should split `unknown` into known but unaddressed widgets and actually unknown widgets, and change the enum order to put yet unaddressed widgets after the important library widgets... we could still do that.

Anyway, this PR just documents the _current_ implementation so let's merge it.

**Pull request merged**